src/Main.elm

```diff
@@ -36,65 +37,68 @@ type alias Model =
     }
 
 
+startGame : Difficulty -> Model
+startGame difficulty =
+    { initialModel
+        | game = difficulty
+        , grid = Grid.fromDimensions (getDimensions difficulty)
+    }
+
+
+initialModel : Model
+initialModel =
+    { grid = Grid.fromDimensions (getDimensions initialDifficulty)
+    , activeCell = Nothing
+    , pressingFace = False
+    , game = initialDifficulty
+    , time = 0
+    , mode = Start
+    , isRightClicked = False
+    , menu = Nothing
+    }
+
+
 type Difficulty
-    = Beginner
-    | Intermediate
-    | Expert
-    | Custom Int Int Int
+    = Custom Int Int Int
+
+
+beginnerSettings : Difficulty
+beginnerSettings =
+    Custom 9 9 10
+
+
+intermediateSettings : Difficulty
+intermediateSettings =
+    Custom 16 16 40
+
+
+expertSettings : Difficulty
+expertSettings =
+    Custom 30 16 99
 
 
 type Menu
     = DifficultyMenu
+    | CustomDifficultyMenu Int Int Int
```

Comment:
what if this were just

```elm
type Menu
    = DifficultyMenu Difficulty
```

Reply:
Yes! That was the idea I was going for. Apologies: I should have made it clear. Sometimes I save my work and push to github as a back up. these are all work in progress commits which i intend to come back and clean up.

Reply:
sorry! didn't mean to rush you